### PR TITLE
Polished crowdflower to training data script (ticket #7)

### PR DESCRIPTION
**Pull request by e-dorigatti**
I also used argparse and added comments and docstrings wherever I had troubles uderstanding what was going on

### CODE REVIEW

File: crowdflower_results_into_training_data.py [hunk 1]
```diff
@@ -7,138 +7,167 @@
...
-    if len(sys.argv) == 4:
-        results = read_full_results(sys.argv[1])
-        #print json.dumps(results, indent=2)
-        maj = set_majority_vote_answer(results)
-        #print json.dumps(maj, indent=2)
-        produce_training_data(maj, sys.argv[2], sys.argv[3])
-    else:
-        print "Usage: %s <CROWDFLOWER_FULL_RESULTS_CSV> <POS_DATA_DIR> <OUTPUT_FILE>" % __file__
-        sys.exit(1)
+    parser = argparse.ArgumentParser()
```
Comment (marfox) [inline crowdflower_results_into_training_data.py]: Good work!
Could you please wrap the creation of the cli parser into a `create_cli_parser()` function?

File: crowdflower_results_into_training_data.py [hunk 2]
```diff
@@ -7,138 +7,167 @@
...
+    parser.add_argument('crowdflower_csv', type=argparse.FileType('r'),
```
Comment (marfox) [inline crowdflower_results_into_training_data.py]: I like this.
Delegate the input file opening to `argparse`

File: crowdflower_results_into_training_data.py [hunk 3]
```diff
@@ -7,138 +7,167 @@
...
+                    for token, tag in tokens:
+                        if token.decode('utf-8') == lines[i][2]:
+                            lines[i][-1] = tag
+
+        output.extend(lines)
+ 
+    return output
+
+
+def main(crowdflower_csv, pos_data_dir, output_file, debug):
```
Comment (marfox) [inline crowdflower_results_into_training_data.py]: the main should return an exit code (0 for success)

File: crowdflower_results_into_training_data.py [hunk 4]
```diff
@@ -7,138 +7,167 @@
...
+                        help='Directory containing the output of treetagger')
+    parser.add_argument('output_file', type=argparse.FileType('w'),
+                        help='Where to store the produced training data')
+    parser.add_argument('--debug', dest='debug', action='store_true')
+    parser.add_argument('-d', dest='debug', action='store_true')
+    parser.add_argument('--no-debug', dest='debug', action='store_false')
+    args = parser.parse_args()
+    assert os.path.exists(args.pos_data_dir)
+
+    main(args.crowdflower_csv, args.pos_data_dir, args.output_file, args.debug)
```
Comment (marfox) [inline crowdflower_results_into_training_data.py]: Please do `sys.exit(main(...))`

**Comment by e-dorigatti** [comment]
Sorry, apparently I forgot to create a new branch so that the fix to ticket 9 ended up here. Will branch now as I found bug ;)

**Pull request merged**